Python/Product/Analysis/Analyzer/ExpressionEvaluator.cs

```diff
@@ -135,10 +136,11 @@ internal class ExpressionEvaluator {
             }
 
             var res = ProjectState.BuiltinModule.GetMember(node, _unit, name);
-            if (createIn != null && !res.Any()) {
-                var refs = createIn.CreateVariable(node, _unit, name, addRef);
+            if (!_unit.ForEval && !res.Any()) {
+                ProjectState.AddDiagnostic(node, _unit, "unknown variable '{0}'".FormatUI(name), LanguageServer.DiagnosticSeverity.Warning, "used-before-assignment");
```

Comment:
It would be nice to consolidate all the error/warning names and descriptions in static class(es). They will be easier to review if they're all in one place.